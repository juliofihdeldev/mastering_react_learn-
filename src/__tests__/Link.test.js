import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Link from '../Link';


test('Link change la classe lorsqu\'on le survole', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>,
  );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
    act(() => {
        /* fire events that update state */
    
        
        // déclenche manuellement le callback
        tree.props.onMouseEnter();
        // re-rendu
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        
        // déclenche manuellement le callback
        tree.props.onMouseLeave();
    });
    // re-rendu
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
 

});