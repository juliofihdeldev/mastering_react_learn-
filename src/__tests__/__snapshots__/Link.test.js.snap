// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link change la classe lorsqu'on le survole 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noopener noreferrer"
>
  Facebook
</a>
`;

exports[`Link change la classe lorsqu'on le survole 2`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noopener noreferrer"
>
  Facebook
</a>
`;

exports[`Link change la classe lorsqu'on le survole 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noopener noreferrer"
>
  Facebook
</a>
`;
